artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java

```diff
@@ -759,6 +769,12 @@ protected void handleReceiveProducerFailCredits(SessionProducerCreditsFailMessag
       handleReceiveProducerFailCredits(message.getAddress(), message.getCredits());
    }
 
+   protected void handleReceiveSlowConsumerKillMessage(DisconnectConsumerWithKillMessage message) {
+      if (message.getNodeID() != null) {
+         setKilled(true);
```

Comment:
since its always called with true, would it make sense to have a `kill()` method instead?